### PR TITLE
fix(batch): Fix table function executor

### PR DESCRIPTION
**Pull request by liurenjie1024**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Currently, the table function executor will accumulate all data chunks before sending out, this may consume a lot of memory if user inputs a large input, for example `range(0, 999999999, 1)`

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]

File: src/batch/src/executor/table_function.rs [hunk 1]
```diff
@@ -50,19 +52,25 @@ impl TableFunctionExecutor {
     async fn do_execute(self: Box<Self>) {
         let dummy_chunk = DataChunk::new_dummy(1);
 
-        let mut builder = self
-            .table_function
-            .return_type()
-            .create_array_builder(self.chunk_size);
-        let mut len = 0;
+        // let mut builder = self
```
Comment (TennyZhuang) [inline src/batch/src/executor/table_function.rs:55]: Remove the comments?

File: src/batch/src/executor/table_function.rs [hunk 2]
```diff
@@ -50,19 +52,25 @@ impl TableFunctionExecutor {
...
+        //     .create_array_builder(self.chunk_size);
+
+        let mut data_chunk_builder =
+            DataChunkBuilder::new(vec![self.table_function.return_type()], self.chunk_size);
         for array in self.table_function.eval(&dummy_chunk).await? {
-            len += array.len();
-            builder.append_array(&array);
+            let len = array.len();
+            for chunk in data_chunk_builder
+                .append_chunk(DataChunk::new(vec![Column::new(array)], Vis::from(len)))
```
Comment (wangrunji0408) [inline src/batch/src/executor/table_function.rs:65]: Perhaps the original match should be kept to handle the multiple return values case.
```suggestion
            let chunk = match array {
                ArrayImpl::Struct(s) => DataChunk::from(s),
                array => DataChunk::new(vec![array.into()], len),
            };
            for chunk in data_chunk_builder.append_chunk(chunk)
```

**Review by xxchan — COMMENTED** [review]
This change won’t help constant range(0, 999999999, 1), because it’s the expr that only returns one large array

**Comment by liurenjie1024** [comment]
> This change won’t help constant range(0, 999999999, 1), because it’s the expr that only returns one large array

Hmm, yes, but this pr breaks this large array into small data chunks so that upstream executor will not need to handle a large input data chunk, this helps when we want to kill the task. It deserves another pr to avoid huge memory allocation.

**Comment by xxchan** [comment]
Get it. I didn't know `append_chunk` can split large chunk into smaller ones 😲

**Pull request merged**